mbed_tools_ci/report_third_party_ip.py

```diff
@@ -0,0 +1,45 @@
+"""Script providing information about licensing and third party IP in order to comply with OpenChain.
+
+The current script uses the sdk provided by the SPDX organisation
+(i.e. https://github.com/spdx/tools-python).
+This SDK only supports version 1.2 of the specification and not 2.1.
+Therefore, some changes will have to be carried out when the later version is
+supported so that third-party IP gets documented as described by the
+specification (i.e. with relationships).
+"""
+import argparse
+import logging
+import sys
+from pathlib import Path
+
+from mbed_tools_ci.utils.logging import set_log_level, log_exception
+from mbed_tools_ci.utils.package_helpers import CurrentProjectMetadataParser
+from mbed_tools_ci.utils.spdx_helpers import SpdxProject
+
+logger = logging.getLogger(__name__)
+
+
+def main() -> None:
+    """Script CLI."""
+    parser = argparse.ArgumentParser(
+        description='Generate licence and third-party IP reports.')
+    parser.add_argument('-o', '--output-dir',
+                        help='Output directory where the files are generated',
+                        required=True,
+                        )
+
+    parser.add_argument("-v", "--verbose", action="count", default=0,
+                        help="Verbosity, by default errors are reported.")
+    args = parser.parse_args()
+    set_log_level(args.verbose)
+
+    try:
+        SpdxProject(CurrentProjectMetadataParser()).generate_tag_value_files(
+            Path(args.output_dir))
+    except Exception as e:
+        log_exception(logger, e)
+        sys.exit(1)
```

Comment:
We should put the `sys.exit` outside of the `main` function - if I ever ran this function in a python interactive interpreter the `sys.exit` would shut the interpreter down, which isn't ideal. Usually you would wrap the `main` call with `sys.exit` and just return an exit code from `main`. Something like the following is the typical style

```
if __name__ == '__main__':
    sys.exit(main())
```